source/common/upstream/upstream_impl.cc

```diff
@@ -830,6 +831,7 @@ ClusterInfoImpl::ResourceManagers::load(const envoy::api::v2::Cluster& config,
   uint64_t max_pending_requests = 1024;
   uint64_t max_requests = 1024;
   uint64_t max_retries = 3;
+  uint64_t max_connection_pools = std::numeric_limits<uint64_t>::max();
```

Comment:
I'm claiming that this is unlimited. I could change the resource manager to just not enforce any limit if it's not set, but I personally think this is good enough.

Reply:
Yeah seems fine to me